aws/resource_aws_kinesis_analyticsv2_application.go

```diff
@@ -0,0 +1,2177 @@
+										Type:     schema.TypeInt,
+										Required: true,
+									},
+									"parallelism_per_kpu": {
+										Type:     schema.TypeInt,
+										Required: true,
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+
+			"sql_application_configuration": {
+				Type:          schema.TypeList,
+				Optional:      true,
+				MaxItems:      1,
+				ConflictsWith: []string{"flink_application_configuration"},
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"inputs": {
+							Type:     schema.TypeList,
+							Optional: true,
+							MaxItems: 1,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"id": {
+										Type:     schema.TypeString,
+										Computed: true,
+									},
+
+									"kinesis_firehose": {
+										Type:     schema.TypeList,
+										Optional: true,
+										MaxItems: 1,
+										Elem: &schema.Resource{
+											Schema: map[string]*schema.Schema{
+												"resource_arn": {
+													Type:         schema.TypeString,
+													Required:     true,
+													ValidateFunc: validateArn,
+												},
+											},
+										},
+									},
+
+									"kinesis_stream": {
+										Type:     schema.TypeList,
+										Optional: true,
+										MaxItems: 1,
+										Elem: &schema.Resource{
+											Schema: map[string]*schema.Schema{
+												"resource_arn": {
+													Type:         schema.TypeString,
+													Required:     true,
+													ValidateFunc: validateArn,
+												},
+											},
+										},
+									},
+
+									"name_prefix": {
+										Type:     schema.TypeString,
+										Required: true,
+									},
+
+									"parallelism": {
+										Type:     schema.TypeList,
+										Optional: true,
+										MaxItems: 1,
+										Elem: &schema.Resource{
+											Schema: map[string]*schema.Schema{
+												"count": {
+													Type:     schema.TypeInt,
+													Required: true,
+												},
+											},
+										},
+									},
+
+									"processing_configuration": {
+										Type:     schema.TypeList,
+										Optional: true,
+										MaxItems: 1,
+										Elem: &schema.Resource{
+											Schema: map[string]*schema.Schema{
+												"lambda": {
+													Type:     schema.TypeList,
+													Required: true,
+													MaxItems: 1,
+													Elem: &schema.Resource{
+														Schema: map[string]*schema.Schema{
+															"resource_arn": {
+																Type:         schema.TypeString,
+																Required:     true,
+																ValidateFunc: validateArn,
+															},
+														},
+													},
+												},
+											},
+										},
+									},
+
+									"schema": {
+										Type:     schema.TypeList,
+										Required: true,
+										MaxItems: 1,
+										Elem: &schema.Resource{
+											Schema: map[string]*schema.Schema{
+												"record_columns": {
```

Comment:
Given that there can be more than one of these and order doesn't seem to be important, `Type: schema.TypeSet` seems more appropriate.

Also, rename to `record_column`, so that HCL would be

```hcl
record_column {
  name     = "name1"
  sql_type = "type1"
}

record_column {
  name     = "name2"
  sql_type = "type2"
}
```